crates/bifrost/src/appender.rs

```diff
@@ -154,37 +160,124 @@ impl Appender {
+                        "New segment detected after {:?}",
+                        total_dur
+                    );
+                }
                 return Ok(loglet);
+            }
+
+            // Okay, tail segment is still sealed
+            let log_metadata_version = Metadata::with_current(|m| m.logs_version());
+            if start.elapsed() > auto_recovery_threshold {
+                tone_escalated = true;
+
+                if error_recovery_strategy >= ErrorRecoveryStrategy::ExtendChainAllowed {
+                    // taking the matter into our own hands
+                    let admin = BifrostAdmin::new(bifrost_inner);
+                    info!(
+                        %sealed_segment,
+                        "[Auto Recovery] Attempting to extend the chain to recover log availability with a new configuration. We waited for {:?} before triggering automatic recovery",
+                        start.elapsed(),
+                    );
+                    if let Err(err) = admin
+                        .seal_and_auto_extend_chain(log_id, Some(sealed_segment))
+                        .log_slow_after(
+                            auto_recovery_threshold / 2,
+                            tracing::Level::INFO,
+                            "Extending the chain with new configuration",
+                        )
+                        .with_overdue(auto_recovery_threshold, tracing::Level::WARN)
+                        .await
+                    {
+                        // we couldn't reconfigure. Let the loop handle retries as normal
+                        info!(
+                            %err,
+                            %log_metadata_version,
+                            "Could not reconfigure the log, perhaps something else beat us to it? We'll check",
+                        );
+                    } else {
+                        info!(
+                            log_metadata_version = %metadata.logs_version(),
+                            "[Auto Recovery] Reconfiguration complete",
+                        );
+                        // reconfiguration successful. Metadata is updated at this point
+                        // Do not fall-through to the backoff sleep
+                        continue;
+                    }
+                }
             } else {
-                let log_version = Metadata::with_current(|m| m.logs_version());
-                debug!(
-                    log_version = %log_version,
-                    "Still waiting for sealing to complete. Elapsed={:?}",
-                    start.elapsed(),
-                );
+                // Holding pattern
+                if tone_escalated {
+                    info!(
+                        %log_metadata_version,
+                        "In holding pattern, still waiting for log reconfiguration to complete. Elapsed={:?}",
+                        start.elapsed(),
+                    );
```

Comment:
Wouldn't we always be in the outer if branch if `tone_escalated == true`?

Reply:
Fixed